x/pool-incentives/keeper/grpc_query.go

```diff
@@ -32,18 +32,30 @@ func (q Querier) GaugeIds(ctx context.Context, req *types.QueryGaugeIdsRequest)
 
 	sdkCtx := sdk.UnwrapSDKContext(ctx)
 	lockableDurations := q.Keeper.GetLockableDurations(sdkCtx)
+	distrInfo := q.Keeper.GetDistrInfo(sdkCtx)
 	gaugeIdsWithDuration := make([]*types.QueryGaugeIdsResponse_GaugeIdWithDuration, len(lockableDurations))
+	totalWeightDec := distrInfo.TotalWeight.ToDec()
+	incentivePercentage := sdk.NewDec(0)
 
 	for i, duration := range lockableDurations {
 		gaugeId, err := q.Keeper.GetPoolGaugeId(sdkCtx, req.PoolId, duration)
 		if err != nil {
 			return nil, status.Error(codes.Internal, err.Error())
 		}
 
+		for _, record := range distrInfo.Records {
+			if record.GaugeId == gaugeId {
+				// Pool incentive % = (gauge_id_weight / sum_of_all_pool_gauge_weight) * 100
+				incentivePercentage = record.Weight.ToDec().Quo(totalWeightDec).MulInt(sdk.NewInt(100))
```

Comment:
I would put `sdk.NewInt(100)` into a var so we don't have to keep allocating memory for it.